compiler/crates/relay-codegen/src/build_ast.rs

```diff
@@ -1850,64 +1850,64 @@ impl<'schema, 'builder, 'config> CodegenBuilder<'schema, 'builder, 'config> {
 
-        let id_prop = ObjectEntry {
-            key: CODEGEN_CONSTANTS.id,
-            value: match request_parameters.id {
-                Some(QueryID::Persisted { id, .. }) => Primitive::RawString(id.clone()),
-                Some(QueryID::External(module_name)) => {
-                    Primitive::JSModuleDependency(JSModuleDependency {
+        let id_prop = if let Some(RequestFormat::ID(ref id) | RequestFormat::Migration(ref id, _)) =
+            request_parameters.request_format
+        {
+            match id {
+                QueryID::Persisted { id, .. } => ObjectEntry {
+                    key: CODEGEN_CONSTANTS.id,
+                    value: Primitive::RawString(id.clone()),
+                },
+                QueryID::External(module_name) => ObjectEntry {
+                    key: CODEGEN_CONSTANTS.id,
+                    value: Primitive::JSModuleDependency(JSModuleDependency {
                         path: *module_name,
                         named_import: None,
                         import_as: None,
-                    })
-                }
-                None => Primitive::Null,
-            },
-        };
-
-        let mut params_object = if let Some(text) = request_parameters.text {
-            vec![
-                ObjectEntry {
-                    key: CODEGEN_CONSTANTS.cache_id,
-                    value: Primitive::RawString(md5(&text)),
-                },
-                id_prop,
-                metadata_prop,
-                name_prop,
-                operation_kind_prop,
-                ObjectEntry {
-                    key: CODEGEN_CONSTANTS.text,
-                    value: Primitive::RawString(text),
-                },
-            ]
-        } else if request_parameters.id.is_some() {
-            vec![
-                id_prop,
-                metadata_prop,
-                name_prop,
-                operation_kind_prop,
-                ObjectEntry {
-                    key: CODEGEN_CONSTANTS.text,
-                    value: Primitive::Null,
+                    }),
                 },
-            ]
+            }
         } else {
-            vec![
-                ObjectEntry {
-                    key: CODEGEN_CONSTANTS.cache_id,
-                    value: Primitive::RawString(md5(operation.name.item.0.lookup())),
-                },
-                id_prop,
-                metadata_prop,
-                name_prop,
-                operation_kind_prop,
-                ObjectEntry {
-                    key: CODEGEN_CONSTANTS.text,
-                    value: Primitive::Null,
-                },
-            ]
+            ObjectEntry {
+                key: CODEGEN_CONSTANTS.id,
+                value: Primitive::Null,
+            }
         };
 
+        let mut params_object = vec![];
+
+        match request_parameters.request_format {
+            Some(
+                RequestFormat::Text(Some(ref text)) | RequestFormat::Migration(_, Some(ref text)),
```

Comment:
Can `RequestFormat` have a method `text()` to get the text from the underlying variant?